_docs/feelgood.md

```diff
@@ -8,4 +8,160 @@ permalink: /feelgood.html
+
+## Architecture
+
+To realize these desired behaviors & user experiences, we used AWS services 
+and deployed the application to the cloud. The application's design is 
+described below.
+
+### User Subscription
+
+* An [S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteHosting.html) 
+  is used to host the subscription site. We configured 
+  [domain redirection for a custom domain](https://www.freecodecamp.org/news/simple-site-hosting-with-amazon-s3-and-https-5e78017f482a/) 
+  and SSL certificates for HTTP/S traffic.
+* The user interacts with the site as they would any other.
+* Upon clicking the `Submit` button, a JavaScript function makes an 
+  [HTTP POST](https://en.wikipedia.org/wiki/POST_%28HTTP%29) request 
+  to the application's API, sending the user's data from the website's 
+  subscription form. The API endpoint is connected to a Lambda function using  
+  [API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-getting-started-with-rest-apis.html).
+* The Lambda function takes the JSON payload delivered by the POST request, 
+  which contains the user's phone number and selected topic. The function 
+  then:
+    * Validates the phone number for correct formatting 
+    * Looks up the [Amazon Resource Name](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) 
+      (ARN) for the selected topic
+    * Invokes the [Simple Notification Service](https://aws.amazon.com/sns/) SDK
+    * Makes a `subscribe()` call to SNS, passing the user's data and topic ARN
+* SNS saves the user's phone number as a subscriber under the specified topic.
+* If the Lambda function processes successfully, it will return a success code 
+  to the API Gateway method, which in turn returns the HTTP 200 status to the 
+  user's browser. The Lambda function also publishes a text message to the user's 
+  provided number, which confirms subscription and welcomes the user.
+
+[View the website code here.](https://github.com/UWB-ACM/feelgood/blob/master/website/)
+
+[View the subscription Lambda function here.](https://github.com/UWB-ACM/feelgood/blob/master/subscribe/lambda_function.py)
+
+Here's a diagram of the system's flow of logic:
+
+![FeelGood subscription mechanism](/assets/feelgood_sub.png)
+
+### User Notification
+
+* To send messages out at regular intervals, we used CloudWatch Events 
+  to create [rules which get invoked on a time schedule](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/Create-CloudWatch-Events-Scheduled-Rule.html).
+* When creating the Event rule, the message publication function 
+  [was selected as the target](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/RunLambdaSchedule.html).
+* Upon being invoked, the Lambda function:
+    * Chooses 4 messages to publish (one for each possible topic)
+    * Looks up the ARN for each topic
+    * Invokes the [Simple Notification Service](https://aws.amazon.com/sns/) SDK
```

Comment:
```suggestion
    * Invokes the SNS \([Simple Notification Service](https://aws.amazon.com/sns/)\) SDK
```

Reply:
Nitpicking but wouldn't 301 rules require it to be the other way? Like: Invokes the [Simple Notification Service](https://aws.amazon.com/sns/) (SNS) SDK

Reply:
Wouldn't it need to be full name then acronym by 301 rules?

Reply:
Sure - I just didn't like how it looked to have the acronym in parentheses immediately followed by a different acronym, but either way is fine by me. Keeping the way it currently is would also be fine since it's easy enough to derive the meaning of SNS in the current context ☺️